react/DeviceCoordinates.tsx

```diff
@@ -77,27 +77,35 @@ const DeviceCoordinates: StorefrontFunctionComponent = () => {
 
     switch (state) {
       case State.PROMPT:
-        icon = <IconLocation />
+        icon = <IconLocation block />
         break
       case State.GRANTED:
-        icon = <IconLocation solid />
+        icon = <IconLocation solid block />
         break
       case State.LOADING:
-        icon = <Spinner size={16} />
+        icon = <Spinner size={16} block />
         break
       case State.DENIED:
-        icon = <IconLocation />
+        icon = <IconLocation block />
         break
       default:
     }
 
-    return loading ? <Spinner size={16} /> : icon
+    if (loading) {
+      icon = <Spinner size={16} />
+    }
```

Comment:
Can't we set the state to `State.LOADING` when the query is loading? This `if` after the `switch` block seems bad.

Reply:
sure!